### PR TITLE
[2.0] add Premake support

### PR DESCRIPTION
**Pull request by Enhex**
Changelog: (Feature): Add Premake tools
Docs: https://github.com/conan-io/docs/pull/XXXX

- [ ] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ ] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one.

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

no documentation so it's been guesswork, let me know if anything is missing.

### CODE REVIEW

**Comment by memsharded** [comment]
Hi @Enhex 

Could you please target this to Conan 1.X (``develop`` branch). All the build system integrations have a new namespace there to not collide with the legacy, and allow a smooth transition from 1.X => 2.0

File: conan/tools/premake/premake.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Premake(object):
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+    # automatically chooses premake action based on used compiler
+    def configure(self):
+        if "Visual Studio" in self.settings.compiler:
```
Comment (memsharded) [inline conan/tools/premake/premake.py:7]: It is likely that ``msvc`` will be the compiler that survive in 2.0. Please try to add support for it too.

**Review by memsharded — COMMENTED** [review]
Some basic testing, at least in ``integration`` (without actually calling premake to build), would be necessary.

Full integration tests are possible adding support in ``conftest.py`` and ``conftest_user.py``, and adding ``pytest.mark.tool_premake`` marks to tests.

**Comment by Enhex** [comment]
I created a new pull request starting off from `develop`:
https://github.com/conan-io/conan/pull/10090

**Pull request closed without merging**